### ISSUE TITLE
Application status not available in Statless k8s Charm when active

### ISSUE THREAD

**Issue by DomFleischmann** [juju/python-libjuju#352]
When deploying a kubernetes charm that is a stateless deployment. The `status` attribute from the value obtained with `model.applications.items()` is empty. When changing the deployment type from `stateless` to `stateful` the status attribute appears.

This can be reproduced with the following bundle:

```
bundle: kubernetes
applications:
  prometheus-k8s:
    charm: 'cs:~charmed-osm/prometheus-k8s-18'
    scale: 1
    options:
      default-target: "mon-k8s:8000"
    storage:
      database: 50M,osm-pv
  grafana-k8s:
    charm: 'cs:~charmed-osm/grafana-k8s-19'
    scale: 1

relations:
  - - 'grafana-k8s:prometheus'
    - 'prometheus-k8s:prometheus'

```

And afterwards executing the following script:

```
from juju.model import Model
from juju import loop


async def test():

    model = Model()

    await model.connect_current()

    for application, app_data in model.applications.items():
        print('{}: {}'.format(application, app_data.status))


def main():
    loop.run(test())


if __name__ == '__main__':
    main()
```

Once the charms are active the script will show nothing for the status of Grafana but active for Prometheus. Grafana is a stateless deployment and Prometheus a Stateful one. If the deployment type of Grafana is changed to stateful the status will be displayed correctly when executing the script.

**Comment by mitechie** [comment]
Thanks for the heads up on this. Pinging in @wallyworld for how we want to handle these types of things.

**Comment by wallyworld** [comment]
The type of deployment (stateful set vs deployment controller) is totally transparent to Juju. I deployed a mariadb charm (stateful) and mediawiki charm (stateless). juju status shows the correct values and the juju database is also correct, eg for a mediawiki charm, the juju db has:

_id: cd1c99d8-3bbb-4e86-8da4-988531da6261:a#mediawiki-k8s
neverset: false
status: active

And running the script showed "active" for both applications. I then tried the bundle from the bug report. Again, all status values were shown as expected, ie juju status and the script output matched.

I did notice though that if I ran the script immediately after deploying the bundle (or individual charms), the status values in the script output could be empty for a short time. I'm not overly familiar with the pylibjuju code, perhaps there's a polling type operation to update a middle tier cache or something.
